File: Frontend/lib/models/news.dart
import 'enums.dart';

class News {
  final int id;
  final String title;
  final String content;
  final bool isPinned;
  final String? imageUrl;
  final DateTime createdDate;

  News({
    required this.id,
    required this.title,
    required this.content,
    required this.isPinned,
    this.imageUrl,
    required this.createdDate,
  });

  factory News.fromJson(Map<String, dynamic> json) {
    return News(
      id: json['id'] as int,
      title: json['title'] as String,
      content: json['content'] as String,
      isPinned: json['isPinned'] as bool,
      imageUrl: json['imageUrl'] as String?,
      createdDate: DateTime.parse(json['createdDate'] as String),
    );
  }
}

class AppNotification {
  final int id;
  final String message;
  final NotificationType type;
  final String? linkUrl;
  final bool isRead;
  final DateTime createdDate;

  AppNotification({
    required this.id,
    required this.message,
    required this.type,
    this.linkUrl,
    required this.isRead,
    required this.createdDate,
  });

  factory AppNotification.fromJson(Map<String, dynamic> json) {
    return AppNotification(
      id: json['id'] as int,
      message: json['message'] as String,
      type: NotificationType.values[json['type'] as int],
      linkUrl: json['linkUrl'] as String?,
      isRead: json['isRead'] as bool,
      createdDate: DateTime.parse(json['createdDate'] as String),
    );
  }
}

File: Backend/Models/News.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PcmBackend.Models
{
    /// <summary>
    /// Bảng tin tức - Tên bảng: [xxx]_News
    /// </summary>
    [Table("056_News")]
    public class News
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string Content { get; set; } = string.Empty;

        /// <summary>
        /// Tin được ghim lên đầu
        /// </summary>
        public bool IsPinned { get; set; } = false;

        /// <summary>
        /// Đường dẫn ảnh minh họa
        /// </summary>
        [MaxLength(500)]
        public string? ImageUrl { get; set; }

        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

        /// <summary>
        /// Người tạo tin (Admin/Staff ID)
        /// </summary>
        public int? CreatedByMemberId { get; set; }

        [ForeignKey("CreatedByMemberId")]
        public virtual Member? CreatedBy { get; set; }
    }
}
